cmd/oras/internal/option/remote.go

```diff
@@ -248,21 +250,27 @@ func (opts *Remote) Credential() auth.Credential {
 }
 
 // NewRegistry assembles a oras remote registry.
-func (opts *Remote) NewRegistry(hostname string, common Common) (reg *remote.Registry, err error) {
+func (opts *Remote) NewRegistry(hostname string, warn func(...interface{}), common Common) (reg *remote.Registry, err error) {
 	reg, err = remote.NewRegistry(hostname)
 	if err != nil {
 		return nil, err
 	}
 	hostname = reg.Reference.Registry
 	reg.PlainHTTP = opts.isPlainHttp(hostname)
+
+	reg.HandleWarning = func(warning remote.Warning) {
+		if _, ok := opts.hasWarned.LoadOrStore(warning.WarningValue, true); ok {
+			warn(warning.Text)
+		}
+	}
```

Comment:
In a copy scenario, should the oras CLI log differently for source or target? I think it can be done by adding a log field to the logger.

Reply:
`NewRegistry` is only used in `oras repo ls` and `oras login` command, so the host name doesn't need to be added here.